llvm/lib/Target/X86/X86CallingConv.td

```diff
@@ -1104,6 +1104,9 @@ def CC_X86 : CallingConv<[
 
 def CSR_NoRegs : CalleeSavedRegs<(add)>;
 
+def CSR_IPRA_32 : CalleeSavedRegs<(add EBP, ESI)>;
```

Comment:
Why ESI, RBX? Are they  frame pointer registers? Could you add tests?

Reply:
They are BasePtr used in x86 frame lowering. https://github.com/llvm/llvm-project/blob/7a086e1b2dc05f54afae3591614feede727601fa/llvm/lib/Target/X86/X86RegisterInfo.cpp#L77 Currently, I don't know how to create an IR to explicitly use BasePtr, if anyone give a hint, it would be appreciated.

Reply:
Found one in code base. Added in [de92c7f](https://github.com/llvm/llvm-project/pull/109597/commits/de92c7f060f9615cffc9368cc11f2990a1728a31)